### PR TITLE
Prepare release 2.0

### PR DESCRIPTION
**Pull request by Jean85**
This PR attempts to start working toward a 2.0, following the roadmap of #338.

@lsmith77 can you create a dedicated branch for 2.x-dev? So I (and others) can target this PR (and following others) to that branch, without polluting master before it gets somewhat stable.

In here I:

 * move to a more PSR-4 friendly dir structure
 * bumped to PHP 7.1
 * bumped deps to allow only deps compatible with Symfony 4
 * dumped Paratest and Assetic because they aren't

This also includes #357

### CODE REVIEW

**Comment by lsmith77** [comment]
here you go.

I wonder if we should not also support Symfony 3.4 with 2.0?

**Comment by Jean85** [comment]
Thanks @lsmith77! I will resolve conflicts now; after that tell me if this is enough to be merged or if you want more stuff included here!

Yes, this PR is allowing Symfony `^3.4|^4.0`!

**Comment by Jean85** [comment]
Ok, Travis build is no longer errored, it's just failing due to only tests.

@lsmith77 if it's ok for you, can you merge this? I would like to start working on it, and first add PHPCSFixer first, so we can create all the (eventual) conflicts at the start, and not in a sparse way.

**Comment by lsmith77** [comment]
let me check with @alexislefebvre on this ..

**Comment by alexislefebvre** [comment]
I can't look at all the changes for now but the summary looks amazing, that's nice!

Creating a 2.0 branch with only these structural changes is probably the best thing to do. Then we'll be able to refactor the bundle by using traits, services or another way.

**Comment by Jean85** [comment]
Great! We'll start working on additional stuff as soon this gets merged!

**Comment by alexislefebvre** [comment]
I'm wondering if we should _move to a more PSR-4 friendly dir structure_ in the current master branch instead of doing this only on the 2.0 branch.

If we use the PSR-4 structure only for the 2.x branch, it means that we'll abandon the 1.x branch because it would require to work for both versions.

If we keep the same structure for both versions, we'll be able to port changes from 1.x to 2.x or backport changes from the 2.x to the 1.x branch.

I have the same question about the splitting of the big `WebTestCase` into several traits.

**Comment by alexislefebvre** [comment]
@Jean85 Did this PR took you several hours? Is it OK for you if we split the files for PSR4 before creating the 2.0 branch?

File: tests/App/cache/prod/AppProdDebugProjectContainer.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?php
```
Comment (alexislefebvre) [inline tests/App/cache/prod/AppProdDebugProjectContainer.php]: @Jean85 This file from cache and all the other cache files should have not been commited.

Comment (lsmith77) [reply tests/App/cache/prod/AppProdDebugProjectContainer.php]: are we missing a .gitignore here?

**Comment by lsmith77** [comment]
@alexislefebvre I personally think its not worth the effort to try and keep 1.x and 2.x in a state so that its easy to move commits from one to the other and it might also impair out ability to do necessary refactorings. To me 1.x is mainly to support 2.7/2.8 until their EOL but I don’t expect people running 2.7/2.8 applications to have much interest in new features let alone changes that risk BC breaks.

File: tests/App/cache/prod/AppProdDebugProjectContainer.php (hunk 1 above)
Comment (alexislefebvre) [reply tests/App/cache/prod/AppProdDebugProjectContainer.php]: Yes, the `.gitignore `file was updated after the commit with all the cache files: https://github.com/liip/LiipFunctionalTestBundle/pull/359/commits/ec55e5c4ecce7320e1e02aef216ae5e432654d53

**Comment by alexislefebvre** [comment]
@Jean85 tests are broken for now, I don't want to merge it in this state. Can you please keep only the structural changes in this PR and open another PR with only the dependencies changes?

**Comment by alexislefebvre** [comment]
In order to fix the `InvalidArgumentException: Unable to find file "@LiipFunctionalTestBundle/Tests/App/DataFixtures/ORM/user.yml".` error, we can use a dummy bundle, as in https://github.com/liip/LiipFunctionalTestBundle/pull/362/commits/29fddaf6e5492e134e68687b14e41b949f46aea0

About the `KERNEL_CLASS` error, it can be set in `phpunit.xml.dist`.

I don't know how to fix the `LogicException: Cannot register "Hautelook\AliceBundle\HautelookAliceBundle" without "Fidry\AliceDataFixtures\Bridge\Symfony\FidryAliceDataFixturesBundle".` error.

**Comment by Jean85** [comment]
@alexislefebvre ok! :+1: I'll be cherry-picking to a new PR for the structural changes so I'll be able to open a consequent one to update deps and fix tests. Doing it now.

**Comment by Jean85** [comment]
Closing this in favor of #365 and #366

**Pull request closed without merging**